### PR TITLE
Upload env vars during next.js deployment.

### PR DESCRIPTION
**Pull request by vladiulianbogdan**
# Pull Request Template

## Type of change

-   [ ] 🍕 New feature
-   [x] 🐛 Bug Fix
-   [ ] 🔥 Breaking change
-   [ ] 🧑‍💻 Improvement
-   [ ] 📝 Documentation Update

## Description

`genezio deploy --env .env` did not work when deploying a Next.js application.

## Checklist

-   [x] My code follows the contributor guidelines of this project;
-   [ ] I have updated the documentation;
-   [ ] I have added tests;
-   [x] New and existing unit tests pass locally with my changes;

### CODE REVIEW

**Review by radu1122 — APPROVED** [review]

File: src/commands/deploy/genezio.ts [hunk 1]
```diff
@@ -327,6 +324,8 @@ export async function genezioDeploy(options: GenezioDeployOptions) {
     for (const frontendUrl of frontendUrls) {
         log.info(colors.cyan(`Frontend URL: ${frontendUrl}`));
     }
+
+    return deployClassesResult;
```
Comment (costinsin) [inline src/commands/deploy/genezio.ts:328]: Why do you return this? I think is not necessary for this change to work, also it does not make sense as this function does multiple things, not just deploying the classes.

File: src/commands/deploy/genezio.ts [hunk 2]
```diff
@@ -530,104 +529,11 @@ export async function deployClasses(
...
-                    );
-                    log.info("");
-                }
-            }
-        }
+        // Deploy environment variables if --env is true
+        const cwd = projectConfiguration.workspace?.backend
+            ? path.resolve(projectConfiguration.workspace.backend)
+            : process.cwd();
+        handleEnvVars(options.env, projectId, projectEnvId, cwd);
```
Comment (costinsin) [inline src/commands/deploy/genezio.ts:536]: Wow! Thanks for cleaning this mess... Now it's much better 🚀

File: src/commands/deploy/nextjs/deploy.ts [hunk 3]
```diff
@@ -96,6 +97,13 @@ export async function nextJsDeploy(options: GenezioDeployOptions) {
     log.info(
         `The app is being deployed at ${colors.cyan(cdnUrl)}. It might take a few moments to be available worldwide.`,
     );
+
+    await handleEnvVars(
+        options.env,
+        deploymentResult.projectId,
+        deploymentResult.projectEnvId,
+        process.cwd(),
+    );
```
Comment (costinsin) [inline src/commands/deploy/nextjs/deploy.ts:106]: Is there a reason why not do this in the `Promise.all` above the log? Or maybe inside `setupEnvironmentVariables`?

File: src/commands/deploy/utils.ts [hunk 4]
```diff
@@ -143,3 +152,95 @@ export async function processYamlEnvironmentVariables(
 
     return newEnvObject;
 }
+
+export async function handleEnvVars(
```
Comment (costinsin) [inline src/commands/deploy/utils.ts:156]: I think this function name is too generic. Without context, I can't tell what it does :))

```suggestion
export async function readAndUploadProjectEnvironment(
```

This is a suggestion, you can replace this with anything else if it sounds better/is shorter.

File: src/commands/deploy/utils.ts [hunk 5]
```diff
@@ -143,3 +152,95 @@ export async function processYamlEnvironmentVariables(
...
+    env: string | undefined,
```
Comment (costinsin) [inline src/commands/deploy/utils.ts:157]:
```suggestion
    envPath: string | undefined,
```

**Review by costinsin — APPROVED** [review]
LGTM

File: src/commands/deploy/genezio.ts (hunk 1 above)
Comment (vladiulianbogdan) [reply src/commands/deploy/genezio.ts:328]: Yes. It's a left over. Thanks

**Pull request merged**